src/string.cr

```diff
@@ -2402,6 +2402,22 @@ class String
     true
   end
 
+  # Returns `self` if not blank, otherwise returns `nil`.
+  #
+  # ```
+  # "a".presence || "default" # => "a"
+  # "".presence || "default"  # => "default"
+  # nil.presence || "default" # => "default"
+  #
+  # data = {} of Symbol => String
+  # data[:missing_key]?.presence || "default" # => "default"
```

Comment:
A hash with symbol keys is a Rubyism and often considered a code smell in Crystal. Let's not use that in an example.
```suggestion
  # config = {"empty" => ""}
  # config["empty"]?.presence || "default"   # => "default"
  # config["missing"]?.presence || "default" # => "default"
```

Reply:
Why is this a code smell? Seems like a valid use case.

Reply:
I'd say using Symbol in Crystal is a code smell, no matter the context.

(I want to eventually remove Symbol from the language, if possible)

Reply:
I considered valid use case too, but is true that is more likely to be a hash of strings.
When a hash is used with an _expected_ list of symbols, then a named tuple can be used.

Reply:
Because you either have dynamic keys in which case you'd use strings (because unlike Ruby, symbols can't be created at runtime) or a fixed set of keys which would better be implemented as a struct/class, named tuple, hash with enum keys. They all improve type safety.